### ISSUE TITLE
Traktor A6 Issue

### ISSUE THREAD

**Issue by korotkov13** [pideck/pideck#97]
Hi! today I bought used Traktor A6 soundcard and tried to use it with pideck.
Unfortunately It didn't recognized as Native Instruments Audio 6 DJ, but only as Traktor Audio 6 DJ.(I don't know exactly why the name of the soundcard is differ. I can suppose that it has the latest firmware, but I'm not sure about it) 
I renamed the device ID in sup_cards.conf and got another issue 
When I connect 1st(left) turntable to A port, it plays the second deck. 
When I connect 2nd(Right) turntable to B port, it didn't play at all,
When I connect 2nd(Right) turntable to master port, it plays the second deck but without Phono, so it is unplayable.
Later I edited T6.asound.conf and after changing to this 
```

pcm.T6_pair1 {
                type plug
                ttable.0.2 1.0
                ttable.1.3 1.0
                slave.pcm T6_duplex
        }

pcm.T6_pair2 {
                type plug
                ttable.0.4 1.0
                ttable.1.5 1.0
                slave.pcm T6_duplex
        }

 pcm.T6_pair3 {
                type plug
                ttable.0.1 1.0
                ttable.1.2 1.0
                slave.pcm T6_duplex
        }
```
It works, BUT! got Another issue with touchscreen controls. When I press PHONO button(to change THRU mode to DVS) on the Left side for the 1st deck, it didn't respond. But it works from Right side controls Left(1st) Turntable.
Could You please help where to dig?

**Comment by danielhjames** [comment]
Hi @korotkov13 thanks for the feedback, please see https://github.com/pideck/pideck/blob/master/touchscreen_controls_L from line 102 and https://github.com/pideck/pideck/blob/master/touchscreen_controls_R from line 92.

**Comment by danielhjames** [comment]
Hi @korotkov13 I have updated the configuration here: f8024b9d2476a6e360af4d2b6bdd0abf04ff2928

What I think might have happened is that your touchscreen buttons were not loaded with the special `--card traktora6` argument due to the difference in device name. It should have been loaded like this:

```
python /usr/share/pideck/touchscreen_controls_L --card traktora6 &
python /usr/share/pideck/touchscreen_controls_R --card traktora6 &
```
Please see https://github.com/pideck/pideck/blob/master/cards_autodetect from line 116.

**Comment by korotkov13** [comment]
Hi @danielhjames! I will try it out tomorrow and let You know!
Thank You for fast reply and helping with this.

**Comment by hoizpi** [comment]
i also have this card!
pair 3 is ttable 0.0 and 1.1 for me

card is called "Traktor Audio 6". NOT "Traktor Audio 6 DJ" .  its without the DJ.
see here --> http://www.pogo.org.uk/~mark/linuxdj/

cv02.5 vinyls arrive this week, i ll test this further!

**Comment by danielhjames** [comment]
Thanks @hoizpi we previously changed the config in https://github.com/pideck/pideck/commit/f8024b9d2476a6e360af4d2b6bdd0abf04ff2928 so it might be there is more than one firmware in these devices.

Please post the output of `cat /proc/asound/cards` so we can see how your Traktor Audio 6 identifies itself. Also your .asoundrc and details of how you connect your turntables would be helpful.

**Comment by hoizpi** [comment]
hi
turntables phono signal gets passed through from thud rumble trx mixer to channel pair 2 and 3 .


i only changed cards_autodetect and sup_cards.conf
pressing phono on the right side turns on THRU-mode und and does not turn off the phono
pressing bypass, which should actually be THRU-mode, does nothing (L and R).

would post screenshot, but cant ssh . no LAN.
so i took a shot with my phone
![image](https://user-images.githubusercontent.com/49638040/56155283-ddb9ae00-5fba-11e9-8bc5-0e5f11be60b3.png)

**Comment by danielhjames** [comment]
Thanks @hoizpi ! Are channels 2 and 3 those marked as A and B on the hardware please?

**Comment by hoizpi** [comment]
hello
just got my gaslampkiller cv02.5 vinyls!!!! IT WORKS!!! atleast the "use timecodes"-part!
got tears in my eyes! much love and respect for releasing this! (edit: love the simplicity!)
is absolute mode supported ? anyway to change from relative mode ?


yes. channel pair 2 and 3 are marked as channel A and B on the hardware!
channel pair 1 is marked as MAIN on the hardware.
pair 1 (MAIN) doesnt have phono preamp- or THRU-switch.
pair 2 and 3 (A and B) do have phono preamp- and THRU-switch.

**Comment by beaucasque** [comment]
Ok guys. @korotkov13 @hoizpi . My pideck setup is working. but having the same problem as everyone with the bypass and PHONO buttons.

My 2 cents as a Dj more than a coder would be that, in any dedicated DJ card like my Traktor Audio 6 Dj and my Rane SL3, the LINE 1 (0.1 and 1.2) should be the master in and out. LINE 2 and 3 are the decks with Preamps.  Also, I've seen some code in a touchscreen_control not present in the other one:

def audioinjector_bypass_button(widget, data):
    if widget.get_active():
       os.system("amixer sset 'Output Mixer HiFi' off")
       os.system("amixer cset name='Line Capture Switch' 0")
       os.system("amixer sset 'Output Mixer Line Bypass' on")
    else:
       os.system("amixer sset 'Output Mixer Line Bypass' off")
       os.system("amixer cset name='Line Capture Switch' 1")
       os.system("amixer sset 'Output Mixer HiFi' on")

As of now, I've kept the T6.asound.conf as it is and changed only the line 117 of cards_autodetect to:

XCARD="-a T6_pair2 -a T6_pair3"


Hope that helps.

**Comment by beaucasque** [comment]
Finally succeeded with the PHONo and ByPASS by changing the numid in both touch_controls:

numid=4,iface=MIXER,name='Direct Thru Channel A'
numid=5,iface=MIXER,name='Direct Thru Channel B'
numid=3,iface=MIXER,name='Internal Clock Validity'
numid=6,iface=MIXER,name='Phono Input Channel A'
numid=7,iface=MIXER,name='Phono Input Channel B'
numid=2,iface=PCM,name='Capture Channel Map'
numid=1,iface=PCM,name='Playback Channel Map'

All is working now for me. The only thing I'd want to do now is have my dicer recognized. Do I only need to edit the /usr/share/pideck/pideck file to do so? and, if so, what needs to be done exactly?

Thank you!

**Comment by danielhjames** [comment]
Hi @beaucasque good to hear you got it working, please let us know if you make a YouTube video of your system so we can add it to our community playlist.

The Dicer support is added with `--dicer hw:Dicer` in the xwax startup command. You can see the changes we made to set this up automatically in the commit https://github.com/pideck/pideck/commit/f30743fb7630cf48e230000b3a913ee2d0f5daf1#diff-9d6f5a3d8fd5247c5d1d673ec712545e

**Comment by beaucasque** [comment]
Yes. I will make a video for sure.  This is so much fun. Way better than my serato experience. Now I only need to tag my music properly. I only see 3 ways of sorting music as of now: bpm, artist, title.. if we could add a "genre" column, it would make dealing with large libraries easier.. or maybe it's there already.

Also, Is there another way to quit than to "F10" and "sudo halt" ?

Thank you!

**Comment by danielhjames** [comment]
Hi @beaucasque it should be straightforward to import and display genre tag information by copying the code for BPM tags. The main issue is lack of screen real estate on smaller devices. There might have to be some toggle to show only certain columns at the same time, or the track title could be truncated to make it fit.

As for shutting down properly, the terminal is the best way in PiDeck version 0.2 but a future release should feature an overlay filesystem so that shutting off the power (on purpose or accidentally) is safe.
